mobile/lib/services/backup.service.dart

```diff
@@ -139,48 +144,71 @@ class BackupService {
     return result;
   }
 
-  Future<List<AssetEntity>> _fetchAssetsAndUpdateLastBackup(
+  Future<Set<BackupCandidate>> _fetchAssetsAndUpdateLastBackup(
     List<AssetPathEntity?> albums,
     List<BackupAlbum> backupAlbums,
-    DateTime now,
   ) async {
-    List<AssetEntity> result = [];
-    for (int i = 0; i < albums.length; i++) {
-      final AssetPathEntity? a = albums[i];
-      if (a != null &&
-          a.lastModified?.isBefore(backupAlbums[i].lastBackup) != true) {
```

Comment:
Why is this removed?
This will greatly increase the time of the background upload each time it's executed for large libraries leading to not uploading stuff at all on iOS and wasting battery on Android as it's triggered very often (after any assets changed on device)

Reply:
Fixed